File: viewer/TweetViewer/Services/LocalMediaFiles.cs
using System.IO;

namespace TweetViewer.Services;

/// <summary>images/&lt;tweet_id&gt;_&lt;idx&gt;.&lt;ext&gt; のローカルパス解決 (拡張子不一致に耐える)。</summary>
public static class LocalMediaFiles
{
    private static readonly string[] ProbeExtensions = { "jpg", "png", "webp", "gif", "jpeg" };

    public static string? Resolve(string imagesDir, string tweetId, int index, string ext)
    {
        var expected = Path.Combine(imagesDir, $"{tweetId}_{index}.{ext}");
        if (File.Exists(expected))
            return expected;
        return ProbeExtensions
            .Select(e => Path.Combine(imagesDir, $"{tweetId}_{index}.{e}"))
            .FirstOrDefault(File.Exists);
    }
}

File: mac/Sources/SubakoCore/Files/LocalMediaFiles.swift
import Foundation

/// images/<tweet_id>_<idx>.<ext> のローカルパス解決 (拡張子不一致に耐える —
/// docs/viewer-features.md §11.2)。
public enum LocalMediaFiles {
    private static let probeExtensions = ["jpg", "png", "webp", "gif", "jpeg"]

    public static func resolve(imagesDir: String, tweetId: String, index: Int, ext: String) -> String? {
        let expected = (imagesDir as NSString).appendingPathComponent("\(tweetId)_\(index).\(ext)")
        if FileManager.default.fileExists(atPath: expected) {
            return expected
        }
        for probe in probeExtensions {
            let candidate = (imagesDir as NSString).appendingPathComponent("\(tweetId)_\(index).\(probe)")
            if FileManager.default.fileExists(atPath: candidate) {
                return candidate
            }
        }
        return nil
    }
}
